### DISCUSSION TITLE
Client-side prototype pollution

### DISCUSSION THREAD

**Discussion by Dinesh-loma** [alpinejs/alpine#4322] [7. Bugs (archived)]
The client-side prototype pollution source `__proto__[property]`  is read from the query string.

example : https://xxxxx.com/xxxx?__proto__[dcb52823]=qblsuvcruj

[CWE-1321: Improperly Controlled Modification of Object Prototype Attributes ('Prototype Pollution')](https://cwe.mitre.org/data/definitions/1321.html)

**Comment by ekwoka** [comment]
There is no code in Alpine that deals with query strings

You are likely looking for https://github.com/livewire/livewire history plugin.

But you'll see here: https://github.com/livewire/livewire/blob/b158c6386a892efc6c5e4682e682829baac1f933/js/plugins/history/index.js#L272

that this is handled.
